### PR TITLE
Translation

### PR DESCRIPTION
**Pull request by etnms**
I added i18n for translations as well as translations files for English and French. I also added more types where they were missing. The users can change the language with a select option in the navbar (the select is basic without any styling at the moment). I added some basic logic to remember the users' preference

### CODE REVIEW

**Comment by yactouat** [comment]
hey @etnms ! sorry was very busy with work at the end of this week, looking at your PR today !

File: frontend/src/i18n.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    translation: translationFR
+  }
+};
+
+i18n
+  .use(initReactI18next) // passes i18n down to react-i18next
+  .init({
+    lng: localStorage.getItem("language") || "en",
+    resources,
+    debug: true,
```
Comment (yactouat) [inline frontend/src/i18n.ts:20]: can we make the `debug` prop depend on an environment variable ? (easier for prod deployments)

File: frontend/src/components/UploadImage.tsx [hunk 2]
```diff
@@ -1,81 +1,105 @@
...
+              break;
+            case "yellow":
+              alert(yellow);
+              break;
+            default:
+              alert(defaultPrediction);
+          }
+        });
     };
+    if (e.target?.files === null) return; // Avoid potential null value
```
Comment (yactouat) [inline frontend/src/components/UploadImage.tsx:80]: thanks for this

**Review by yactouat — APPROVED** [review]
this is great, thanks a lot ! (I left a few comments, but none is blocking)

File: frontend/src/i18n.ts (hunk 1 above)
Comment (etnms) [reply frontend/src/i18n.ts:20]: I believe so yes

**Pull request merged**